### PR TITLE
fix(server): storage usage calculation for motion photos

### LINKED ISSUE #8511: [BUG] Deleting android motion photos cause storage usage to go negative
### The bug

When a user uploads a motion photo in the `MP` or `MVIMG` format, and then permanently deletes the photo, the amount of storage space "used" by the user decreases beyond the size of the `jpg`. The size of the video portion of the motion photo seems to be double counted.

Reproduction steps:
1. Create a user account with a storage quota.
2. Login as the user.
3. Upload a motion photo from the web GUI, here's an example https://files.catbox.moe/b59ftp.jpg
4. Permanently delete the motion photo.
5. Observe that your "used storage" value becomes negative

Here's a video demonstrating the reproduction steps:

https://github.com/immich-app/immich/assets/44907675/f1665c6a-7a02-4b6e-9655-571aaa762b36

The "used storage" value goes back to 0 correctly if I change the user account's storage quota.

### The OS that Immich Server is running on

Fedora Linux 37

### Version of Immich Server

v1.100.0

### Version of Immich Mobile App

v1.100.0

### Platform with the issue

- [X] Server
- [X] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
version: '3.8'

#
# WARNING: Make sure to use the docker-compose.yml of the current release:
#
# https://github.com/immich-app/immich/releases/latest/download/docker-compose.yml
#
# The compose file on main may not be compatible with the latest release.
#

name: immich

services:
  immich-prometheus:
    container_name: immich_prometheus
    ports:
      # this exposes the default port for Prometheus so you can interact with it
      - 9090:9090
    image: prom/prometheus
    volumes:
      # the Prometheus configuration file - a barebones one is provided to get started
      - ./prometheus.yml:/etc/prometheus/prometheus.yml
      # a named volume defined in the bottom of the Compose file; it can also be a mounted folder
      - prometheus-data:/prometheus

  immich-grafana:
    container_name: immich_grafana
    command: ['./run.sh', '-disable-reporting'] # this is to disable Grafana's telemetry
    ports:
      - 3000:3000
    image: grafana/grafana
    volumes:
      # stores your pretty dashboards and panels
      - grafana-data:/var/lib/grafana
 
  immich-server:
    container_name: immich_server
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    command: ['start.sh', 'immich']
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - /etc/localtime:/etc/localtime:ro
      - ${EXTERNAL_PATH}:/usr/src/app/external
    env_file:
      - .env
    ports:
      - 2283:3001
    depends_on:
      - redis
      - database
    restart: always

  immich-microservices:
    container_name: immich_microservices
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    extends: # uncomment this section for hardware acceleration - see https://immich.app/docs/features/hardware-transcoding
      file: hwaccel.transcoding.yml
      service: nvenc # set to one of [nvenc, quicksync, rkmpp, vaapi, vaapi-wsl] for accelerated transcoding
    command: ['start.sh', 'microservices']
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - /etc/localtime:/etc/localtime:ro
      - ${EXTERNAL_PATH}:/usr/src/app/external
    env_file:
      - .env
    depends_on:
      - redis
      - database
    restart: always

  immich-machine-learning:
    container_name: immich_machine_learning
    # For hardware acceleration, add one of -[armnn, cuda, openvino] to the image tag.
    # Example tag: ${IMMICH_VERSION:-release}-cuda
    image: ghcr.io/immich-app/immich-machine-learning:${IMMICH_VERSION:-release}-cuda
    extends: # uncomment this section for hardware acceleration - see https://immich.app/docs/features/ml-hardware-acceleration
      file: hwaccel.ml.yml
      service: cuda # set to one of [armnn, cuda, openvino, openvino-wsl] for accelerated inference - use the `-wsl` version for WSL2 where applicable
    volumes:
      - model-cache:/cache
    env_file:
      - .env
    restart: always

  redis:
    container_name: immich_redis
    image: registry.hub.docker.com/library/redis:6.2-alpine@sha256:51d6c56749a4243096327e3fb964a48ed92254357108449cb6e23999c37773c5
    restart: always

  database:
    container_name: immich_postgres
    image: registry.hub.docker.com/tensorchord/pgvecto-rs:pg14-v0.2.0@sha256:90724186f0a3517cf6914295b5ab410db9ce23190a2d9d0b9dd6463e3fa298f0
    environment:
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_DB: ${DB_DATABASE_NAME}
    volumes:
      - pgdata:/var/lib/postgresql/data
    restart: always

volumes:
  pgdata:
  model-cache:
  prometheus-data:
  grafana-data:
```

### Your .env content

```Shell
# You can find documentation for all the supported env variables at https://immich.app/docs/install/environment-variables

# The location where your uploaded files are stored
UPLOAD_LOCATION=<redacted>

# The Immich version to use. You can pin this to a specific version like "v1.71.0"
IMMICH_VERSION=release

# Connection secret for postgres. You should change it to a random password
DB_PASSWORD=<redacted>

IMMICH_METRICS=true
EXTERNAL_PATH=<redacted>

# The values below this line do not need to be changed
###################################################################################
DB_HOSTNAME=immich_postgres
DB_USERNAME=postgres
DB_DATABASE_NAME=immich

REDIS_HOSTNAME=immich_redis
LOG_LEVEL=debug
```

### Reproduction steps

```bash
See above
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Ynng**
Closes #8511, originally part of #8512.

Excluded all external library assets from storage usage, including `isExternal = false` ones.

Stop erroneously updating storage usage during asset deletion for external library assets.

Update storage usage during motion photo video extraction.

### CODE REVIEW

File: server/src/repositories/user.repository.ts [hunk 1]
```diff
@@ -117,11 +118,15 @@ export class UserRepository implements IUserRepository {
...
+    // we can't use parameters with getQuery, hence the template string
     const subQuery = this.assetRepository
       .createQueryBuilder('assets')
       .select('COALESCE(SUM(exif."fileSizeInByte"), 0)')
+      .leftJoin('assets.library', 'library')
       .leftJoin('assets.exifInfo', 'exif')
-      .where('assets.ownerId = users.id AND NOT assets.isExternal')
+      .where('assets.ownerId = users.id')
+      .andWhere('NOT assets.isExternal')
+      .andWhere(`library.type != '${LibraryType.EXTERNAL}'`)
```
[the remark this answers is not in the document]
Comment (Ynng) [reply server/src/repositories/user.repository.ts:129]: Motion photo video assets in external libraries are `isExternal = false` but do not count towards storage usage, hence checking the library type.

I just kept `NOT assets.isExternal` around as I thought it's more robust/correct.

[the remark this answers is not in the document]
Comment (jrasm91) [reply server/src/repositories/user.repository.ts:129]: I think we should restructure this query to only include library type is upload. No need to negate all these external conditions. Let's simplify it to just be "quota for the upload library".

**Review by jrasm91 — APPROVED** [review]
I think this looks good. Basically
- whenever we use assetRepository.create we need to update the quota
- whenever we update the quota we need to check if the event is for an upload asset

At some point we might want to extract this asset create logic into a util method, so the same logic isn't scattered around everywhere, but I think it looks good for now.

File: server/src/services/asset.service.ts [hunk 2]
```diff
@@ -391,7 +391,9 @@ export class AssetService {
     }
 
     await this.assetRepository.remove(asset);
-    await this.userRepository.updateUsage(asset.ownerId, -(asset.exifInfo?.fileSizeInByte || 0));
+    if (asset.library.type !== LibraryType.EXTERNAL) {
```
[the remark this answers is not in the document]
Comment (Ynng) [reply server/src/services/asset.service.ts]: It's done

File: server/src/repositories/user.repository.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ynng) [reply server/src/repositories/user.repository.ts]: Ok I made the change

**Review by danieldietzler — APPROVED** [review]
Good job! Thank you :)

**Pull request merged**